Практическая работа 3

potatoe61 opened the pull request:
Добрый день! Вставляю обязательное "Я студент высшего образования программы онлайн-бакалавриата, трек Android-разработки." Чутка накосячил с первым ПР, заслал не туда:)

potatoe61 committed:
Практическая работа 3

AleksandrIlinskii commented:
Я только запустил программу, он мне пишет - нет архивов, но при этом сразу - выберите архив. А какой? 
<img width="692" alt="image" src="https://github.com/user-attachments/assets/cde819cd-26f5-420c-86cf-37e8e334cfbe">

AleksandrIlinskii commented:
В требованиях есть пункт - нельзя давать создавать архив с пустым именем 
<img width="727" alt="image" src="https://github.com/user-attachments/assets/ed92fee9-70ff-44e5-bb8b-900d5310569d">

AleksandrIlinskii commented:
Также имеются пункты - нельзя создавать заметки с пустым именем или содержимым
<img width="804" alt="image" src="https://github.com/user-attachments/assets/22dbe875-05e1-483a-bb3d-93dc9c8de4aa">

AleksandrIlinskii commented on src/main/kotlin/NotesApp.kt:
```diff
@@ -0,0 +1,54 @@
+data class Archive(val id: Int, val name: String)
```
Каждый data class, да и любой класс в принципе должен быть в отдельном файле

AleksandrIlinskii commented on src/main/kotlin/NotesApp.kt:
```diff
@@ -0,0 +1,54 @@
+data class Archive(val id: Int, val name: String)
+data class Note(val id: Int, val name: String, val content: String)
```
Если аргументов больше чем 1 - каждый новый пишется с новой строки
<img width="272" alt="image" src="https://github.com/user-attachments/assets/69d619f2-1966-4c10-a59d-632bf51cf613">

AleksandrIlinskii commented on src/main/kotlin/NotesApp.kt:
```diff
@@ -0,0 +1,54 @@
+data class Archive(val id: Int, val name: String)
+data class Note(val id: Int, val name: String, val content: String)
+
+class NotesApp {
```
Название класса должно отражать происходящее внутри него. У тебя называется "Приложение для заметок", но работа идет и с архивами, хотя про них в названии ничего нет

AleksandrIlinskii commented on src/main/kotlin/NotesApp.kt:
```diff
@@ -0,0 +1,54 @@
+            println("Нет доступных архивов.")
+        } else {
+            archives.forEach { archive ->
+                println("${archive.id}. ${archive.name}")
```
В данном случае лучше переопределить toString у класса Archive (как на скрине) и просто писать как на втором скрине 
<img width="345" alt="image" src="https://github.com/user-attachments/assets/4dccdddc-b1c2-41bb-aecb-cf5b0cdd7c80">
<img width="351" alt="image" src="https://github.com/user-attachments/assets/6da1f544-7a17-476d-8f7f-b7bd3d06bb1f">

AleksandrIlinskii commented on src/main/kotlin/NotesApp.kt:
```diff
@@ -0,0 +1,54 @@
+    fun viewNote(archiveId: Int, noteIndex: Int) {
+        val selectedNote = notes[archiveId]?.get(noteIndex - 1)
+        selectedNote?.let {
+            println("Содержимое заметки '${it.name}': ${it.content}")
```
Аналогично с архивами тут можно сделать toString для заметок

AleksandrIlinskii commented on src/main/kotlin/UserInterface.kt:
```diff
@@ -0,0 +1,80 @@
+        val input = readLine() ?: return true
+        val option = input.toIntOrNull()
+
+        return when {
```
котлин позволяет создавать переменные в конструкторе when + это позволяет не писать option в каждой ветке 
<img width="594" alt="image" src="https://github.com/user-attachments/assets/9ad183fb-6fad-42c7-8aa6-6e50c94c44b3">

AleksandrIlinskii commented on src/main/kotlin/UserInterface.kt:
```diff
@@ -0,0 +1,80 @@
+        val input = readLine() ?: return true
+        val option = input.toIntOrNull()
+
+        return when {
```
А вот тут такой номер не прокатит, потому что ты кроме option проводишь еще другие проверки

potatoe61 committed:
Практическая работа 3
Попытка 2. Добавил обработчики событий при вводе пустых названий архивов и заметок. Содержание тоже теперь не может быть пустым
